opal/util/if.h

```diff
@@ -37,9 +37,7 @@
 #include <netinet/in.h>
 #endif
 
-#ifndef IF_NAMESIZE
-#define IF_NAMESIZE 32
-#endif
+#define OPAL_IF_NAMESIZE 32
```

Comment:
It would be probably worth putting a comment here explaining a little history, and where the magic value of 32 came from, etc. (I know that such an explanation didn't already exist, but given that we just have to have a conversation digging up the past, it means that there really *should* be a comment here explaining what/why/etc.).

Reply:
I'm not actually sure where the magic value of 32 comes from myself. I read in the other thread:

> I could swear we recently had a debate about namesize and the value we should use to ensure that our array was at least as large as whatever the system used. I thought we resolved it be picking a size that was bigger than anything we might encounter (in Linux or whatever OS flavor we encounter). Is there some reason that fix wasn't adequate (I can't immediately find the PR where all that was discussed)?

I do not know the actual history behind this value though.

Reply:
I was referring to your commit message on #6849 

> In cases where opal/util/if.h is included but net/if.h is not, IF_NAMESIZE will be 32. If net/if.h is included on Linux systems, IF_NAMESIZE will be 16. This can cause a mismatch when using the same macro macro on a system. To avoid this error case, we include this file in util/if.h.

It would be good to include a little history here that we previously just #define'd `IF_NAMESIZE` to 32 blah blah blah... that was a bad idea because blah blah blah... so now we #define `OPAL_IF_NAMESIZE` to avoid the issue blah blah blah...

That kind of text.  Something to help our Future Selves when someone looks at this code 6-12 months from now.

Reply:
Got it, updating